Source/Other/Extensions/DateExtension.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  DateExtention.swift
+//  Sequel Ace
+//
+//  Created by James on 3/11/2020.
+//  Copyright © 2020 Sequel-Ace. All rights reserved.
+//
+
+import Foundation
+
+extension Date {
+	
+	public func format(_ format: String, locale: Locale, timeZone: TimeZone) -> String {
+		let formatter = DateFormatter.mediumStyleFormatter
+		
+		formatter.dateFormat = format
+		formatter.locale = locale
+		formatter.timeZone = timeZone
+		
+		return formatter.string(from: self)
+	}
+	
+	public func format(_ format: String, timeZone: TimeZone) -> String {
+		return self.format(format, locale: .current, timeZone: timeZone)
+	}
+	
+	public func format(_ format: String, locale: Locale) -> String {
+		return self.format(format, locale: locale, timeZone: .current)
+	}
+	
+	public func format(_ format: String) -> String {
+		return self.format(format, locale: .current, timeZone: .current)
+	}
+}
+
+@objc extension NSDate {
```

Comment:
I don't think you need this as long as you put the `objc` or `objcMembers` in front of the `extension Date`? String vs NSString is something else, but here it should work... :)

Reply:
Seems that @objc or objcMembers is just for classes, not extensions

Reply:
And if you mark the functions as @objc?